src/LearningRateSchedulers.jl

```diff
@@ -313,7 +306,20 @@ function calculate_timestep!(
 
     y_mean = get_obs(ekp)
 
-    Φ = [0.5 * norm(inv_sqrt_Γ * (g[:, j] - reshape(y_mean, :, 1)))^2 for j in 1:J]
+    # efficiently compute [0.5 * norm(inv(sqrt(Γ)) * (g_j - y))^2 for j in 1:J]
```

Comment:
I don't really understand what is happening in the loop below with `shift` but I trust that this comment is accurate.

Reply:
First, I have checked that the errors are within `~10^-13`, which actually comes due to the error in taking the uneccessary matrix square root in the old code.


Second, as a brief note to explain. 
- We can rewrite the `norm(sqrt(A)*(g_j-y))^2` as `(g_j - y)' * A * (g_j-y)`, 
- We compute `A*(g_j - y)` and dot this with `(gj-y)`,
- Rather than doing this for every `j` separately do it in one sweep with matrix algebra

The matrix product looks gross, as we also replace a matrix `A` with block matrices `[A1,A2,A3...]`. And therefore when performing a multiplication, `Ax` we have to multiply `A1*x[idx1,:]` and `A2*x[idx,:]`, thus we need to keep track of indices